packages/trader/src/App/Components/Elements/Modals/ServicesErrorModal/services-error-modal.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Button, Modal } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { getTitle } from './constants';
+import AuthorizationRequiredModal from './authorization-required-modal.jsx';
+import InsufficientBalanceModal from './insufficient-balance-modal.jsx';
+import CompanyWideLimitExceededModal from './company-wide-limit-exceeded-modal.jsx';
+
+type TServicesError = {
+    code: string;
+    message: string;
+    type?: string;
+};
+
+type TPropServicesErrorModel = {
+    is_virtual?: boolean;
+    is_visible: boolean;
+    is_logged_in?: boolean;
+    onConfirm: () => void;
+    services_error: TServicesError;
+};
+
+const ServicesErrorModal = ({
+    is_virtual,
+    is_visible,
+    is_logged_in,
+    onConfirm,
+    services_error,
+}: TPropServicesErrorModel) => {
+    const { code, message, type } = services_error;
+
+    if (!code || !message) return null;
+
+    switch (code) {
+        case 'AuthorizationRequired':
+            return (
+                <AuthorizationRequiredModal
+                    is_logged_in={is_logged_in}
+                    is_visible={is_visible}
+                    toggleModal={onConfirm}
+                />
+            );
+        case 'InsufficientBalance':
+            return (
+                <InsufficientBalanceModal
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    // TODO: This component needs to be TS migrated so error isn't there
+                    is_virtual={is_virtual}
+                    is_visible={is_visible}
+                    message={message}
+                    toggleModal={onConfirm}
+                />
+            );
+        case 'CompanyWideLimitExceeded':
+            return <CompanyWideLimitExceededModal is_visible={is_visible} onConfirm={onConfirm} />;
+        default:
+            return (
+                <Modal is_open={is_visible} small title={getTitle(type)} toggleModal={onConfirm}>
+                    <Modal.Body>{message}</Modal.Body>
```

Comment:
Just to clarify: does `message` have auto translation?  I mean you use `localize` to translate `OK`, but do not applied it to `message`.

Reply:
It wasnt wrapped `services-error-modal.jsx` so I kept it the way it is. Perhaps, message comes translated from back end.

Reply:
I just asked to be sure) If the message was translated somewhere in BE than no problem